llvm/lib/Transforms/Vectorize/SandboxVectorizer/SandboxVectorizer.cpp

```diff
@@ -56,31 +108,13 @@ bool SandboxVectorizerPass::runImpl(Function &LLVMF) {
     return false;
   }
 
-  sandboxir::Context Ctx(LLVMF.getContext());
-  // Create SandboxIR for `LLVMF`.
-  sandboxir::Function &F = *Ctx.createFunction(&LLVMF);
-  // Create the passes and register them with the PassRegistry.
-  sandboxir::PassRegistry PR;
-  auto &BottomUpVecPass = static_cast<sandboxir::FunctionPass &>(
-      PR.registerPass(std::make_unique<sandboxir::BottomUpVec>()));
-
-  sandboxir::FunctionPassManager *PM = nullptr;
-  if (UserDefinedPassPipeline == DefaultPipelineMagicStr) {
-    // Create the default pass pipeline.
-    PM = &static_cast<sandboxir::FunctionPassManager &>(PR.registerPass(
-        std::make_unique<sandboxir::FunctionPassManager>("pm")));
-    PM->addPass(&BottomUpVecPass);
-  } else {
-    // Create the user-defined pipeline.
-    PM = &PR.parseAndCreatePassPipeline(UserDefinedPassPipeline);
-  }
-
   if (PrintPassPipeline) {
```

Comment:
oh, is this the only reason `SandboxVectorizerPass` has a reference to the region pass manager? I think ideally only the `BoUpVectorizer` has a reference to it, and we can call `BottomUpVecPass->printPipeline()`, which forwards to `RPM->printPipeline()`

Reply:
Yeah, I'm fine with `BottomUpVec` owning the region pass manager instead. I considered it and took the lazy way out because it made it easier to print the pipeline here :)

I'll change it.

Reply:
Done. I'm not sure I'm completely happy with the pattern where the RegionPassManager object is actually owned by the PassRegistry, but I don't think it will cause any problems either and if we go with it we can always change it afterwards.

Reply:
yeah, all the non-owning pointers make me sus